starport/pkg/cosmosutil/path.go

```diff
@@ -0,0 +1,71 @@
+package cosmosutil
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+// ChainHomeRoot is the root dir for spn chain homes
+const (
+	ChainHomeRoot = "spn"
+	gentxPath     = "config/gentx/gentx.json"
+	genesisPath   = "config/genesis.json"
+)
+
+// IsChainHomeExist checks if a home with the provided launchID already exist
+func IsChainHomeExist(launchID uint64) (string, bool, error) {
+	home, err := ChainHome(launchID)
+	if err != nil {
+		return home, false, err
+	}
+
+	if _, err := os.Stat(home); os.IsNotExist(err) {
+		return home, false, nil
+	}
+	return home, true, err
+}
+
+// ChainHome returns the default home dir used for a chain from SPN
+func ChainHome(launchID uint64) (string, error) {
+	home, err := os.UserHomeDir()
+	return filepath.Join(home, ChainHomeRoot, strconv.FormatUint(launchID, 10)), err
+}
+
+// GenesisPath returns the default genesis path into the home dir
+func GenesisPath(home string) string {
+	return filepath.Join(home, genesisPath)
+}
+
+// Gentx returns the default gentx path into the home dir
+func Gentx(home string) string {
+	return filepath.Join(home, gentxPath)
+}
```

Comment:
Are they necessary? `chain.Chain` implements `GenesisPath()` and `GentxPath()`

Reply:
The `Chain` structure is created from a cloned repo into a temporary folder inside the `Blockchain` struct. I didn't use the `Blockchain` struct, only the `Builder`. We can move the `chain *chain.Chain` to inside the `Builder` struct, but we need to make a lot of chances into all commands logic. Because now the chain info is not fetched for the `Join` command